tools/github-actions/audit/src/reports.ts

```diff
@@ -0,0 +1,179 @@
+  core.debug(response);
+  const { nbVulnerabilities, highestSeverityFound } = computeTotalsForSeverities((severity) => reportJson.metadata.vulnerabilities[severity]);
+  return Object.values(reportJson.vulnerabilities)
+    .reduce<OtterAuditReport>((currentVulnerabilities, advisory) =>
+      updateReportWithVulnerability(currentVulnerabilities, severityThreshold, {
+        severity: advisory.severity,
+        moduleName: advisory.name,
+        overview: typeof advisory.via[0] === 'string' ? advisory.via[0] : advisory.via[0].title
+      })
+    , {errors: [], warnings: [], nbVulnerabilities, highestSeverityFound} as OtterAuditReport);
+};
+
+/**
+ * Format the response from npm audit in a common interface that will be used to build the
+ * report summary
+ * @param response
+ * @param severityThreshold
+ */
+export const computeNpmReport: OtterAuditReporter = (response: string, severityThreshold: Severity) => {
+  core.info('Computing Report for Npm');
+  const reportJson = JSON.parse(response) as NPMAuditReportV2.Audit | NPMAuditReportV1.Audit;
+  if ('auditReportVersion' in reportJson && reportJson.auditReportVersion === 2) {
+    return computeNpmV2Report(response, severityThreshold);
+  } else {
+    return computeNpmV1Report(response, severityThreshold);
+  }
+};
+
+/**
+ * Format the response from yarn 1 npm audit in a common interface that will be used to build the
+ * report summary
+ * @param response
+ * @param severityThreshold
+ */
+export const computeYarn1Report: OtterAuditReporter = (response: string, severityThreshold: Severity) => {
+  core.info('Computing Report for Yarn 1');
+  const reports = response.split('\n').filter(a => !!a).map((report) => JSON.parse(report)) as YarnAudit.AuditResponse[];
+  const reportSummary = reports.filter((report): report is YarnAudit.AuditSummary => report.type === 'auditSummary')[0];
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- Wrong typings in 'audit-types'
+  const { nbVulnerabilities, highestSeverityFound } = computeTotalsForSeverities((severity) => (reportSummary.data as any).vulnerabilities[severity]);
+  return reports.filter((report): report is YarnAudit.AuditAdvisory => report.type === 'auditAdvisory')
+    .reduce<OtterAuditReport>((currentVulnerabilities, advisory) =>
+      updateReportWithVulnerability(currentVulnerabilities, severityThreshold, {
+        severity: advisory.data.advisory.severity,
+        moduleName: advisory.data.advisory.module_name,
+        overview: advisory.data.advisory.overview
+      })
+    , {errors: [], warnings: [], nbVulnerabilities, highestSeverityFound} as OtterAuditReport);
+};
+
+/**
+ * Format the response from yarn 2 and 3 npm audit in a common interface that will be used to build the
+ * report summary
+ * @param response
+ * @param severityThreshold
+ */
+export const computeYarn3Report: OtterAuditReporter = (response: string, severityThreshold: Severity) => {
+  core.info('Computing Report for Yarn 3');
```

Comment:
Could you retrieve the yarn version (2 or 3) from anywhere?
So that we don't confuse people using yarn 2.

Reply:
I've updated to `Yarn 2 and 3`